core/src/main/scala/io/projectglow/transformers/normalizevariants/VariantNormalizer.scala

```diff
@@ -67,92 +68,372 @@ private[projectglow] object VariantNormalizer extends GlowLogging {
-              if (splitToBiallelic) {
-                splitVC(vc)
-              } else {
-                Stream(vc)
-              }
-
-            val isFromSplit = vcStreamAfterMaybeSplit.length > 1
-
-            val vcStreamAfterMaybeNormalize = if (doNormalize) {
-              vcStreamAfterMaybeSplit.map(vc =>
-                VariantNormalizer.normalizeVC(vc, refGenomeDataSource.get))
-            } else {
-              vcStreamAfterMaybeSplit
-            }
-
-            vcStreamAfterMaybeNormalize.map(
-              variantContextToInternalRowConverter.convertRow(_, isFromSplit)
-            )
-
-          case None => Stream(row)
+        it.map { row =>
+          val isFromSplit = row.getBoolean(splitFromMultiallelicColumnIdx)
+          internalRowToVariantContextConverter.convert(row) match {
 
+            case Some(vc) =>
+              variantContextToInternalRowConverter
+                .convertRow(VariantNormalizer.normalizeVC(vc, refGenomeDataSource.get), isFromSplit)
+            case None => row
+          }
         }
       }
+
+      SQLUtils.internalCreateDataFrame(df.sparkSession, rddAfterNormalize, schema, false)
+
+    } else {
+      dfAfterMaybeSplit
     }
 
-    SQLUtils.internalCreateDataFrame(df.sparkSession, mapped, schema, false)
+    dfAfterMaybeNormalize
 
   }
 
   /**
-   * Splits a vc to bi-alleleic using gatk splitting function
+   * Generates a new DataFrame by splitting the variants in the input DataFrame similar to what vt decompose does
    *
-   * @param vc
-   * @return a Stream of split vc's
+   * @param variantDf
+   * @return dataframe of split variants
    */
-  private def splitVC(vc: VariantContext): Stream[VariantContext] = {
-    GATKVariantContextUtils
-      .splitVariantContextToBiallelics(
-        vc,
-        false,
-        GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL,
-        false
+  @VisibleForTesting
+  private[normalizevariants] def splitVariants(variantDf: DataFrame): DataFrame = {
+
+    if (variantDf.schema.fieldNames.contains("attributes")) {
+      // TODO: Unflattened INFO field splitting
+      logger.warn(
+        "The variant splitting transformer does not split unflattened INFO fields within the attributes column.")
+    }
+
+    // Update splitFromMultiAllelic column, add INFO_OLD_MULTIALLELIC column (see vt decompose)
+    // and posexplode alternateAlleles column
+    val dfAfterAltAlleleSplit = variantDf
+      .withColumn(
+        splitFromMultiAllelicField.name,
+        when(size(col(alternateAllelesField.name)) > 1, lit(true)).otherwise(lit(false))
+      )
+      .withColumn(
+        infoFieldPrefix + oldMultiallelicFieldName,
+        when(
+          col(splitFromMultiAllelicField.name),
+          concat(
+            col(contigNameField.name),
+            lit(":"),
+            expr(s"${startField.name} + 1"),
+            lit(":"),
+            concat_ws("/", col(s"${refAlleleField.name}"), col(s"${alternateAllelesField.name}"))
```

Comment:
No need to use string interpolation here. Just `col(refAlleleField.name)`.

Reply:
changed